djl-spring-boot-mxnet-autoconfigure/src/main/java/ai/djl/spring/configuration/DjlAutoConfiguration.java

```diff
@@ -47,22 +48,51 @@
 
     @Bean
     public ZooModel<?, ?> model() throws MalformedModelException, ModelNotFoundException, IOException {
-        MxModelZooType modelType = properties.getMxModelZooType();
-        if(modelType == null) {
-            LOG.warn("Model loader is not defined. Using default SSD loader {}", MxModelZooType.SSD);
-            modelType = MxModelZooType.SSD;
+        ApplicationType applicationType = properties.getApplicationType();
+        EngineType engineType = properties.getEngineType();
+        String artifactId = properties.getArtifactId();
+        Map<String, String> filter = properties.getModelFilter();
+        Map<String, Object> arguments = properties.getArguments();
+        Class<?> inputClass = properties.getInputClass();
+        if (inputClass == null) {
+            LOG.warn("Input class is not defined. Using default: BufferedImage");
+            inputClass = BufferedImage.class;
         }
-        return modelType.getLoader().loadModel(properties.getModelCriteria(), new ProgressBar());
+        Class<?> outputClass = properties.getOutputClass();
+        if (outputClass == null) {
+            LOG.warn("Input class is not defined. Using default: DetectedObjects");
+            outputClass = DetectedObjects.class;
+        }
+
+        Criteria.Builder<?, ?> builder = Criteria.builder().setTypes(inputClass, outputClass);
+        if (applicationType != null) {
+            builder.optApplication(applicationType.application());
+        }
+        if (engineType != null) {
+            builder.optEngine(engineType.engineName());
+        }
+        if (filter != null) {
+            builder.optFilters(filter);
+        }
+        if (arguments != null) {
+            builder.optArguments(arguments);
+        }
+        if (artifactId != null) {
+            builder.optModelZooName(artifactId);
+        }
+
+        return ModelZoo.loadModel(builder.build());
```

Comment:
What does it load if nothing is specified in the properties? the test shows `zoo/ssd/tiny/pikachu/0.0.1/ssd-0030.params` but I am unclear how this selection is made. If it is the default, then it makes sense to be explicit (in javadocs) what is going to happen by default.

Reply:
1.  Only inputClass and outPutClass are required. in 0.3.0 only BufferedImage is support, outputClass is either Classifications or DetectedObjects depends on the application, 0.4.0 will support URL and File as input
2. There isn't a default model defined, it search all the model zoo included in the project and return return the first matching model. And it may not be a deterministic search. In the demo ai.djl.zoo and ai.djl.mxnet:mxnet-model-zoo are included, so pikachu detection model is return if no further filter is configured.
3. You can now override translator behavior by defining arguments, for example change threshold value.

Reply:
On #3, it was my request, I had it on my list to integrate. 
With respect to search being non-deterministic: if pickachu model is returned when filters don't match, then let's reflect it in the documentation. If pickachu selection is non-deterministic, then let's assign something by default in the configuration so that users know what to expect.

Reply:
Do you want to put in javadocs or update the `DjlConfigurationProperties` with default app type? Not a hard req, but ideally a deterministic result should happen when loading empty config.